api/utils/sshutils/ssh.go

```diff
@@ -250,3 +260,263 @@ func KeysEqual(ak, bk ssh.PublicKey) bool {
+	var ppk strings.Builder
+	ppk.WriteString(fmt.Sprintf("PuTTY-User-Key-File-3: %v\n", keyType))
+	ppk.WriteString(fmt.Sprintf("Encryption: %v\n", encryptionType))
+	ppk.WriteString(fmt.Sprintf("Comment: %v\n", fileComment))
+	// chunk the Teleport-formatted public key into 64-character length lines
+	chunkedPublicKey := chunkString(teleportPublicKey, 64)
+	ppk.WriteString(fmt.Sprintf("Public-Lines: %v\n", len(chunkedPublicKey)))
+	for _, r := range chunkedPublicKey {
+		ppk.WriteString(fmt.Sprintf("%s\n", r))
+	}
+	// chunk the PPK-formatted private key into 64-character length lines
+	chunkedPrivateKey := chunkString(string(ppkPrivateKeyBase64), 64)
+	ppk.WriteString(fmt.Sprintf("Private-Lines: %v\n", len(chunkedPrivateKey)))
+	for _, r := range chunkedPrivateKey {
+		ppk.WriteString(fmt.Sprintf("%s\n", r))
+	}
+	ppk.WriteString(fmt.Sprintf("Private-MAC: %v\n", macString))
+
+	// convert string to bytes and return
+	return []byte(ppk.String()), nil
+}
+
+// chunkString converts a string into a []string with chunks of size chunkSize
+// used to split base64-encoded strings across multiple lines with an even width
+func chunkString(s string, chunkSize int) []string {
+	if len(s) == 0 {
+		return nil
+	}
+	if chunkSize >= len(s) {
+		return []string{s}
+	}
+	chunks := make([]string, 0, (len(s)-1)/chunkSize+1)
+	currentLen := 0
+	currentStart := 0
+	for i := range s {
+		if currentLen == chunkSize {
+			chunks = append(chunks, s[currentStart:i])
+			currentLen = 0
+			currentStart = i
+		}
+		currentLen++
+	}
+	chunks = append(chunks, s[currentStart:])
+	return chunks
+}
+
+// getRFC4251Mpint returns a stream of bytes representing a mixed-precision integer (a big.Int in Go)
+// prepended with a big-endian uint32 expressing the length of the data following.
+// This is the 'mpint' format in RFC4251 Section 5 (https://datatracker.ietf.org/doc/html/rfc4251#section-5)
+func getRFC4251Mpint(n *big.Int) ([]byte, error) {
+	buf := new(bytes.Buffer)
+	b := n.Bytes()
+	// RFC4251: If the most significant bit would be set for a positive number, the number MUST be preceded by a zero byte.
+	if b[0]&0x80 > 0 {
+		b = append([]byte{0}, b...)
+	}
+	// write a uint32 with the length of the byte stream to the buffer
+	err := binary.Write(buf, binary.BigEndian, uint32(len(b)))
+	if err != nil {
+		return nil, trace.Wrap(fmt.Errorf("failed to write length uint32 to buffer: %T", err))
+	}
+	// write the byte stream representing of the rest of the integer to the buffer
+	err = binary.Write(buf, binary.BigEndian, b)
+	if err != nil {
+		return nil, trace.Wrap(fmt.Errorf("failed to write data to buffer: %T", err))
+	}
+	return buf.Bytes(), nil
+}
+
+// getRFC4251String returns a stream of bytes representing a string prepended with a big-endian unit32
+// expressing the length of the data following.
+// This is the 'string' format in RFC4251 Section 5 (https://datatracker.ietf.org/doc/html/rfc4251#section-5)
+func getRFC4251String(data []byte) ([]byte, error) {
```

Comment:
As mentioned before, writing to a `bytes.Buffer` won't return an error, so you can probably remove the error return from this.

Reply:
Funny thing is that I wrote all this without any error checking initially and the linter was perfectly content. I should've realised that I meant I was doing it right all along :)